compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -443,17 +443,10 @@ trait Applications extends Compatibility {
     /** The function's type after widening and instantiating polytypes
      *  with TypeParamRefs in constraint set
      */
-    @threadUnsafe lazy val methType: Type = {
-      def rec(t: Type): Type = {
-        t.widen match{
-          case funType: MethodType => funType
-          case funType: PolyType => 
-            rec(instantiateWithTypeVars(funType))
-          case tp => tp
-        }
-      }
```

Comment:
Even if this isn't currently needed with the restriction on type parameter clauses, I don't think it's a bad idea to keep the more general version of this code.